### PR TITLE
Spec change for Issue 18474

### PR DESCRIPTION
**Pull request by RazvanN7**
See: https://github.com/dlang/dmd/pull/8098

### CODE REVIEW

**Comment by wilzbach** [comment]
Nope only git commit --amend to get a new Sha and then force-pushing it.

**Comment by wilzbach** [comment]
@RazvanN7 solved the merge conflicts and squashed.

File: spec/struct.dd [hunk 1]
```diff
@@ -927,31 +927,39 @@ immutable struct C2
...
-// Unlike B2, this example also does not work due to additional restrictions on shared.
-shared struct D2
+struct B
 {
-    // error: non-shared method S.__postblit is not callable using a shared object
-    S a;
+    A a;
 }
----
 )
```
Comment (wilzbach) [inline spec/struct.dd]: @RazvanN7 this is the reason for the failure:

```
---
    import std.stdio;
    shared struct A
    {
        this(this)
        {
            import std.stdio : writeln;
            writeln("the shared postblit was called");
        }
    }
    
    struct B
    {
        A a;
    }
    ) // < - notice the parenthesis here
    
    void main()
    {
        shared B b1;
        auto b2 = b1;
    }
    
---
__stdin.d(15): Error: declaration expected, not `)`
__stdin.d(21): Error: unrecognized declaration
```

**Comment by RazvanN7** [comment]
@wilzbach fixed

**Pull request merged**